src/modules/ekf2/EKF/mag_control.cpp

```diff
@@ -67,8 +50,25 @@ void Ekf::controlMagFusion()
 		_num_bad_flight_yaw_events = 0;
 	}
 
-	if (_control_status.flags.mag_fault || !_control_status.flags.tilt_align) {
+	// When operating without a magnetometer and no other source of yaw aiding is active,
+	// yaw fusion is run selectively to enable yaw gyro bias learning when stationary on
+	// ground and to prevent uncontrolled yaw variance growth
+	// Also fuse zero heading innovation during the leveling fine alignment step to keep the yaw variance low
+	if (_params.mag_fusion_type >= MAG_FUSE_TYPE_NONE
```

Comment:
I'm thinking about what happens when _control_status.flags.ev_yaw or _control_status.flags.gps_yaw. This change fixes the mag use case, but if using EV or GPS yaw don't we end up with the same problem during tilt alignment?

Should we move the fusion of zero innovation yaw measurements into the

 	if (!_control_status.flags.tilt_align) {

conditional branch in Ekf::controlFusionModes()

Reply:
@priseborough `gps_yaw` and `ev_yaw` fusions cannot start before tilt alignment complete. This means that the fake fusion will always be executed here. However, I agree that we might want to refactor a bit to make it clearer (we should also refactor `fuseHeading` to extract a `fuseFakeHeading` function). We can do that in a follow-up PR.

Reply:
eg: https://github.com/priseborough/PX4-Autopilot/commit/a14496670aeeb5c25172ec43ad975484e395cddd

Reply:
Yes you're right. However this is a special case that is there to stop covariances blowing out during tilt alignmnet so maybe we should move it out of the mag fusion control path anyway